### DISCUSSION TITLE
Escaped `\+` not working when used along with `|`?

### DISCUSSION THREAD

**Discussion by tapyu** [helix-editor/helix#8336] [Troubleshooting] [answered]
### Summary

I apologize if that is a typo on my side, but it seems that the scaped `\+`does not work when it is used with `|`.

### Reproduction Steps

Consider the following text and the following `select` regex pattern

![first](https://github.com/helix-editor/helix/assets/22801918/f5394580-9672-4307-92f7-b9e0cd62697e)

It should match `C` and `C++`, shouldn't???

On the other hand, for some reason, we can get this done if we swap the regex matches:

![second](https://github.com/helix-editor/helix/assets/22801918/7dbb7c56-789c-46b3-a221-3787cb43d368)

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-09-19T02:07:25.418 helix_view::editor [ERROR] Failed to initialize the LSP for `source.md` { cannot find binary path }
```

</details>

### Platform

Linux

### Terminal Emulator

kitty 0.21.2 created by Kovid Goyal

### Helix Version

helix 23.05 (7f5940be)

**Comment by kirawi** [comment, marked as answer]
I'm pretty sure it's because C gets matched before C++ in the first example. `C\b|C\+\+` would work for example.

**Reply by tapyu** [reply]
I didn't work either, but I think you find the "issue". That behavior is the same as other Regex's matching in other software. Therefore that is not a helix bug.

**Reply by pascalkuthe** [reply]
its simply a matter of ordering alternations correctly, `C|C\+\+` does work.

**Reply by gabydd** [reply]
I think ``C\+\+|C`` is probably what Pascal meant

**Reply by pascalkuthe** [reply]
ah right sory I copied the wrong thing
